backend-rust/src/indexer.rs

```diff
@@ -5028,16 +5046,82 @@ impl PreparedPayDayBlock {
     }
 }
 
+/// Represents the payday pool commission rates to passive delegators captured
+/// from the `get_passive_delegation_info` node endpoint.
+struct PreparedPassiveDelegationPaydayCommissionRates {
+    transaction_commission:  Option<i64>,
+    baking_commission:       Option<i64>,
+    finalization_commission: Option<i64>,
+}
+
+impl PreparedPassiveDelegationPaydayCommissionRates {
+    fn prepare(passive_delegation_status: Option<PassiveDelegationStatus>) -> anyhow::Result<Self> {
+        Ok(Self {
+            transaction_commission: passive_delegation_status.as_ref().map(|status| {
+                i64::from(u32::from(PartsPerHundredThousands::from(
+                    status.commission_rates.transaction,
+                )))
+            }),
+
+            baking_commission: passive_delegation_status.as_ref().map(|status| {
+                i64::from(u32::from(PartsPerHundredThousands::from(status.commission_rates.baking)))
+            }),
+
+            finalization_commission: passive_delegation_status.as_ref().map(|status| {
+                i64::from(u32::from(PartsPerHundredThousands::from(
+                    status.commission_rates.finalization,
+                )))
+            }),
+        })
+    }
+
+    async fn save(
+        &self,
+        tx: &mut sqlx::Transaction<'static, sqlx::Postgres>,
+    ) -> anyhow::Result<()> {
+        // The fields `transaction_commission`, `baking_commission`, and
+        // `finalization_commission` are either all `Some` or all `None`.
+        if let (
```

Comment:
Our pattern does not really fit well for terminating quickly. We should probably look into adding some common trait to the Prepared structs, such that we can provide a conditional as to when we want to save.

Reply:
After discussions:
Moving checks up the functions will create bigger, more unclear top-level functions (that handle many cases). e.g.:
```
if let (Some(transaction_commission),
            Some(baking_commission),
            Some(finalization_commission),
        ) = (self.transaction_commission, self.baking_commission, self.finalization_commission)
        {

       self.passive_delegation_payday_commission_rates.save(tx).await?;
}
```
We keep the current behavior. Large `Prepared structs` can benefit from introducing a flag to specify whether to save the data to the database (as Lasse suggested).